src/master-playlist-controller.js

```diff
@@ -852,42 +852,37 @@ export class MasterPlaylistController extends videojs.EventTarget {
    * @return {TimeRange} the current time
    */
   setCurrentTime(currentTime) {
-    let buffered = Ranges.findRange(this.tech_.buffered(), currentTime);
-
-    if (!(this.masterPlaylistLoader_ && this.masterPlaylistLoader_.media())) {
-      // return immediately if the metadata is not ready yet
-      return 0;
-    }
-
-    // it's clearly an edge-case but don't thrown an error if asked to
-    // seek within an empty playlist
-    if (!this.masterPlaylistLoader_.media().segments) {
-      return 0;
-    }
-
-    // In flash playback, the segment loaders should be reset on every seek, even
-    // in buffer seeks. If the seek location is already buffered, continue buffering as
-    // usual
-    // TODO: redo this comment
-    if (buffered && buffered.length) {
-      return currentTime;
-    }
+    const media = this.masterPlaylistLoader_ && this.masterPlaylistLoader_.media();
+    const segments = media && media.segments || [];
+    const buffered = this.tech_.buffered();
+    const ctBuffered = Ranges.findRange(this.tech_.buffered(), currentTime);
+
+    // Start segment loaders at the new location if:
+    // * master media is ready
+    // * master media has segments
+    // * It is not too early for tech to have a buffer
+    // * The seek requested is not already buffered
+    if (media && segments.length  && buffered.length && !ctBuffered.length) {
```

Comment:
The bug seemed to be that we `resetEverything` and `abort` right after before playback starts, but after things have started loading. To work around this we check that we check wether we have anything buffered at all, and if we don't then we do not reset or abort.

Reply:
should this be `media && segments.length  && !buffered.length && !ctBuffered.length)`?

Reply:
no because if we don't have any `tech` buffer yet then its likely that we haven't even started playback yet.

Reply:
it's possible, given the comment on previous line 868, that this change will break Flash playback